src/masks/InteractionMasks.tsx

```diff
@@ -86,34 +76,33 @@ export default function InteractionMasks<R>({
   onRowsUpdate,
   scrollToCell
 }: InteractionMasksProps<R>) {
-  const [selectedPosition, setSelectedPosition] = useState<Position>(() => {
+  const [selectedPosition, setSelectedPosition] = useState<SelectCellState | EditCellState>(() => {
     if (enableCellAutoFocus && document.activeElement === document.body && columns.length > 0 && rows.length > 0) {
-      return { idx: 0, rowIdx: 0 };
+      return { idx: 0, rowIdx: 0, status: 'SELECT' };
     }
-    return { idx: -1, rowIdx: -1 };
+    return { idx: -1, rowIdx: -1, status: 'SELECT' };
   });
   const [copiedPosition, setCopiedPosition] = useState<Position & { value: unknown } | null>(null);
   const [draggedPosition, setDraggedPosition] = useState<DraggedPosition | null>(null);
-  const [firstEditorKeyPress, setFirstEditorKeyPress] = useState<string | null>(null);
-  const [isEditorEnabled, setIsEditorEnabled] = useState(false);
   const selectionMaskRef = useRef<HTMLDivElement>(null);
 
-  // Focus on the selection mask when the selected position is changed
+  // Focus on the selection mask when the selected position is changed or the editor is closed
   useEffect(() => {
-    if (selectedPosition.rowIdx === -1 || selectedPosition.idx === -1) return;
+    if (selectedPosition.rowIdx === -1 || selectedPosition.idx === -1 || selectedPosition.status === 'EDIT') return;
     selectionMaskRef.current?.focus();
   }, [selectedPosition]);
 
-  // Focus on the selection mask after the editor is closed
-  useEffect(() => {
-    if (isEditorEnabled) return;
-    selectionMaskRef.current?.focus();
-  }, [isEditorEnabled]);
-
   useEffect(() => {
     return eventBus.subscribe('SELECT_CELL', selectCell);
   });
 
+  const closeEditor = useCallback(() => {
+    setSelectedPosition(({ idx, rowIdx }) => ({ idx, rowIdx, status: 'SELECT' }));
+  }, []);
+
+  // close editor when scrolling
+  useEffect(() => closeEditor, [scrollTop, scrollLeft, closeEditor]);
```

Comment:
This should be moved in the editor container.
- every time we scroll, we re-render InteractionMasks, and the effect cleanup function (`closeEditor`) will be called
- `closeEditor` then updates selectedPosition with a new state. Since it's an new object, it'll trigger a re-render

That doesn't sound too good. 🤔

Reply:
Moved it back to EditorContainer